<a href="https://colab.research.google.com/github/JulioHenri/Tecnicas-de-amostragem/blob/main/amostragem_aleatoria.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importando bibliotecas

In [ ]:
import pandas as pd
import numpy as np

# Carregando e lendo o banco de dados

In [ ]:
dados = pd.read_csv('HR_Engagement_Sat_Sales_UpdatedV4.0.csv') #carregando arquivo csv
dados.head() #lendo os 5 primeiros registros do dataset

ID  ... Sensor_Proximity(1-highest/10-lowest)
0   1  ...                                     9
1   2  ...                                     8
2   3  ...                                     8
3   4  ...                                     6
4   5  ...                                    10

[5 rows x 62 columns]

In [ ]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 62 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   ID                                     14999 non-null  int64  
 1   Name                                   14997 non-null  object 
 2   Department                             14999 non-null  object 
 3   GEO                                    14999 non-null  object 
 4   Role                                   14999 non-null  object 
 5   Rising_Star                            21 non-null     float64
 6   Will_Relocate                          14999 non-null  int64  
 7   Critical                               480 non-null    float64
 8   Trending Perf                          14999 non-null  int64  
 9   Talent_Level                           14999 non-null  int64  
 10  Validated_Talent_Level                 14999 non-null  int64  
 11  Pe

Podemos notar acima que há 14999 registros e 62 colunas. Também há várias atributos com valores faltantes que serão analisados posteriormente.

# Exercícios

Exercício 1.1

a. unidade de pesquisa: Se o empregado continua ou não na empresa

b. população: Funcionários de uma empresa

c. instrumento de coleta de dados: Sistema de cadastro da empresa

d. unidade respondente: Empresas na área de Recursos Humanos

e. possível sistema de referência: 

f. unidade amostral mais provável: Apenas funcionários com salário "baixo"

g. unidades amostrais alternativas: Funcionários do departamento de Finanças e  os que não foram promovidos nos últimos 5 anos

Exercício 1.5 

a. Árvores em uma floresta: Nesse caso, iria fazer a amostra aleatória sistemática pois ela é de rápida e fácil execução e dessa maneira poderíamos selecionar as diversas árvores e separa-las por características específicas e depois de ter feito isso, e depois escolher um número igual de árvores para cada grupo. Sendo assim, tentando ao máximo ter uma amostra que represente a população. Poderíamos fazer um estudo que prevê se uma árvore será desmatada ou não, baseado em suas características, tempo de vida e localidade em que se encontra.

b. Crianças abaixo de 5 anos e que tiveram sarampo:
Como a população em questão já é bem ninchada (específica), a amostragem aleatória simples é o mais indicado pela sua facilidade de aplicação. A variável a ser estudada poderia ser as sequelas que cada criança teve e a gravidade que a doença causou, para isso, iriamos ter que saber quanto tempo ele esteve no hospital, quanto tempo esteve com a doença, e características físicas e mentais de cada criança

c. Operários em indústrias têxteis: Amostragem estratificada é o melhor tipo de forma de obter uma amostra para esta população, pois separamos a população em grupos e subgrupos e após isso, podemos usar amostragem aleatória simples para cada grupo especificado. Através de entrevistas, poderíamos estudar a satisfação dos funcionários em relação ao trabalho e concluir cada avaliação dos empregados entre três classes (bom, médio ou ruim). Para obter essas informações, os questionários teriam informações pessoais de cada operário (idade, raça, gênero) e informações relacionadas ao trabalho como quanto tempo trabalha nessa indústria, se tem parentes trabalhando no mesmo setor e etc.



# Amostragem Simples

Simplesmente selecionar alguns valores do banco de dados de maneira aleatória. Nesse caso irei selecionar 40% dos meus registros.

In [ ]:
amostra = dados.sample(frac=0.40)
amostra.head()

ID  ... Sensor_Proximity(1-highest/10-lowest)
3275  3276  ...                                     9
1347  1348  ...                                     8
6616  6617  ...                                    10
3529  3530  ...                                     8
5761  5762  ...                                     7

[5 rows x 62 columns]

In [ ]:
amostra.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6000 entries, 3275 to 13286
Data columns (total 62 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   ID                                     6000 non-null   int64  
 1   Name                                   5999 non-null   object 
 2   Department                             6000 non-null   object 
 3   GEO                                    6000 non-null   object 
 4   Role                                   6000 non-null   object 
 5   Rising_Star                            6 non-null      float64
 6   Will_Relocate                          6000 non-null   int64  
 7   Critical                               179 non-null    float64
 8   Trending Perf                          6000 non-null   int64  
 9   Talent_Level                           6000 non-null   int64  
 10  Validated_Talent_Level                 6000 non-null   int64  
 11  

Obtendo a amostra, temos 6000 registros, selecionando 40% da população de maneira aleatória. 

# Amostragem Sistemática

Os elementos da população serão colocados em uma lista e cada enésimo elemento da lista é escolhido (sistematicamente) e irá compor a amostra.

In [ ]:
seed = np.random.choice(10,1) #selecionando número aleatório de 1 a 10
seed

array([8])

In [ ]:
indices = np.arange(0, len(dados), seed)
indices

array([    0,     8,    16, ..., 14976, 14984, 14992])

No primeiro parâmetro do código acima eu seleciono em qual número o array deve se iniciar, no segundo em que número ele deve parar e por último, o intervalo com que os números devem ser escolhidos.


**Obs1**: o método 'np.arange' retorna um array (lista) de valores.

**Obs2**: len(dados) retorna a quantidade total de registros do DataFrame **dados**.

In [ ]:
amostra = dados.loc[indices]

No código acima eu localizei, dentro dos meus dados, somente os indíces que correspondem ao meu array 'indices', explicado anteriormente.

**OBS**: Índice de um DataFrame é o primeiro valor do registro da esquerda para direita, que não tem nenhuma label em cima.

In [ ]:
amostra.head()

ID  ... Sensor_Proximity(1-highest/10-lowest)
0    1  ...                                     9
8    9  ...                                     7
16  17  ...                                     9
24  25  ...                                     7
32  33  ...                                     9

[5 rows x 62 columns]

In [ ]:
amostra.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1875 entries, 0 to 14992
Data columns (total 62 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   ID                                     1875 non-null   int64  
 1   Name                                   1875 non-null   object 
 2   Department                             1875 non-null   object 
 3   GEO                                    1875 non-null   object 
 4   Role                                   1875 non-null   object 
 5   Rising_Star                            2 non-null      float64
 6   Will_Relocate                          1875 non-null   int64  
 7   Critical                               59 non-null     float64
 8   Trending Perf                          1875 non-null   int64  
 9   Talent_Level                           1875 non-null   int64  
 10  Validated_Talent_Level                 1875 non-null   int64  
 11  Per

Como podemos observar, a nossa amostra gerada varia a quantidade de registros de acordo com o número gerado aleatoriamente, selecionando assim, de maneira sistemática.